src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,336 @@
+            magneto.Z = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(4));
+
+            if (OutputBitMode == OutputBitMode.Output16bit)
+            {
+                // From the documentation range is from 32760 which does represent 4912 µT
+                // result of 4912.0f / 32760.0f
+                magneto *= 0.1499389499389499f;
+            }
+            else
+            {
+                // result of 4912.0f / 8192.0f
+                magneto *= 0.599609375f;
+            }
+            return magneto;
+
+        }
+
+        /// <summary>
+        /// Read the magnetometer with bias correction and can wait for new data to be present
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </remarks>
+        /// <param name="waitForData">true to wait for new data</param>
+        /// <param name="timeout">timeout for waiting the data, ignored if waitForData is false</param>
+        /// <returns>The data from the magnetometer</returns>
+        public Vector3 ReadMagnetometer(bool waitForData, TimeSpan timeout)
+        {
+            var magn = ReadMagnetometerWithoutCorrection(waitForData, timeout);
+            magn.X *= MagnetometerHardwareAdjustmeent.X;
+            magn.Y *= MagnetometerHardwareAdjustmeent.Y;
+            magn.Z *= MagnetometerHardwareAdjustmeent.Z;
+            magn -= MagnetometerBias;
+            return magn;
+        }
+
+        /// <summary>
+        /// <![CDATA[
+        /// Get or set the device self test mode. 
+        /// If set to true, this creates a magnetic field
+        /// Once you read it, you will have the results of the self test
+        /// 14-bit output(BIT=“0”)  
+        ///          | HX[15:0]        | HY[15:0]        | HZ[15:0] 
+        /// Criteria | -50 =< HX =< 50 | -50 =< HY =< 50 | -800 =< HZ =< -200 
+        /// 16-bit output(BIT=“1”)  
+        ///          | HX[15:0]          | HY[15:0]          | HZ[15:0] 
+        /// Criteria | -200 =< HX =< 200 | -200 =< HY =< 200 | -3200 =< HZ =< -800 
+        /// ]]>
+        /// </summary>
+        public bool MageneticFieldGeneratorEnabled
+        {
+            get => _selfTest;
+
+            set
+            {
+                byte mode = value ? (byte)0b01000_0000 : (byte)0b0000_0000;
+                WriteRegister(Register.ASTC, mode);
+                _selfTest = value;
+            }
+        }
+
+        /// <summary>
+        /// Select the measurement mode
+        /// </summary>
+        public MeasurementMode MeasurementMode
+        {
+            get => _measurementMode;
+
+            set
+            {
+                byte mode = (byte)((byte)value | ((byte)_outputBitMode << 4));
+                WriteRegister(Register.CNTL, mode);
+                _measurementMode = value;
```

Comment:
same comment about reset (same question about others)

Reply:
Yes, I adjusted that as well and the OutputBitMode one.